cdc/sink/manager.go

```diff
@@ -146,7 +146,7 @@ func (t *tableSink) FlushRowChangedEvents(ctx context.Context, resolvedTs uint64
 		return t.manager.flushBackendSink(ctx)
 	}
 	resolvedRows := t.buffer[:i]
-	t.buffer = t.buffer[i:]
+	t.buffer = append(make([]*model.RowChangedEvent, 0, len(t.buffer[i:])), t.buffer[i:]...)
```

Comment:
Could you add a unit test to make sure the buffer length is under a reasonable value?